crypto/defaults.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 1995-2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/thread_once.h"
+#include "internal/cryptlib.h"
+#include "internal/e_os.h"
+
+#if defined(_WIN32)
+
+/**
+ * @brief The directory where OpenSSL is installed.
+ */
+static char openssldir[MAX_PATH];
+
+/**
+ * @brief The directory where OpenSSL engines are located.
+ */
+
+static char enginesdir[MAX_PATH];
+
+/**
+ * @brief The directory where OpenSSL modules are located.
+ */
+static char modulesdir[MAX_PATH];
+
+/**
+ * @brief Get the list of Windows registry directories.
+ *
+ * This function retrieves a list of Windows registry directories.
+ *
+ * @return A pointer to a char array containing the registry directories.
+ */
+static char *get_windows_regdirs(char *dst, LPCWSTR valuename)
+{
+    DWORD keysize;
+    DWORD ktype;
+    HKEY hkey;
+    LSTATUS ret;
+    LPCWCH tempstr = NULL;
+    char *retval = NULL;
+
+    ret = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                       TEXT("SOFTWARE\\Wow6432Node\\OpenSSL"), 0,
```

Comment:
Why `SOFTWARE\\Wow6432Node\\OpenSSL...` ? Why not just `SOFTWARE\\OpenSSL....`? That Wow6432Node thing is for 32 bit apps running on 64 bit Windows.

Reply:
NSIS builds installers into a 32 bit binary for compatibility (even though its perfectly capable of installing 64 bit binaries).  as such the installer writes registry keys to the WOW6432Node reflector, which is where we have to look them up

Reply:
OK, now I get it. Is potential 32 bit OpenSSL build able to see the `SOFTWARE\\Wow6432Node\\OpenSSL` keys or it sees just `SOFTWARE\\OpenSSL`? In such case it would have to be conditional on whether this is a 64 bit or 32 bit build.